packages/ui/src/AdminAddMCQ.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useState } from "react";
+import { Sheet, SheetContent, SheetTrigger } from "./shad/ui/sheet";
+import { Button } from "./shad/ui/button";
+import { Input } from "./shad/ui/input";
+import { Card, CardDescription, CardTitle } from "./shad/ui/card";
+import { MCQQuestion, Problem } from "@prisma/client";
+import { createMCQ, deleteMCQ } from "../../../apps/web/components/utils";
+
+interface AdminAddMCQProps extends Problem {
+  mcqQuestions: MCQQuestion[];
+}
+
+const AdminAddMCQ = ({ problem }: { problem: AdminAddMCQProps }) => {
+  const [mcqs, setMcqs] = useState<MCQQuestion[]>([]);
+  const [ExistingMCQs, setExistingMCQs] = useState<MCQQuestion[]>(problem.mcqQuestions);
+  const [question, setQuestion] = useState<string>("");
+  const [options, setOptions] = useState<string[]>([]);
+  const [option, setoption] = useState<string>("");
+  const [correctOption, setCorrectOption] = useState<string>("");
+  function handleAddOption() {
+    setOptions([...options, ...option.split(",")]);
+    setoption("");
+  }
+  function handleRemoveOption(option: string) {
+    setOptions(options.filter((o) => o !== option));
+  }
+  function handleAddMCQ() {
+    const data = {
+      id: problem.id + "-" + (mcqs.length + ExistingMCQs.length),
+      question: question,
+      options: options,
+      correctOption: correctOption,
+      problemId: problem.id,
+    };
+    setMcqs([...mcqs, data]);
+    setQuestion("");
+    setOptions([]);
+    setoption("");
+    setCorrectOption("");
+  }
+  function handleRemoveMCQ(id: string) {
+    setMcqs(mcqs.filter((mcq) => mcq.id !== id));
+  }
+  function handleSubmit() {
```

Comment:
@ShubhamMewara Please add a try/catch block for this async function.

Reply:
if you go inside utils.tsx where I have created createMCQ function there is try-catch and since it is the only function I am calling I don't think we need try-catch here